From ba59bab7530cf98b40b7c4e1c8f86dc3d7f2b47f Mon Sep 17 00:00:00 2001
From: David Ung <davidu@nvidia.com>
Date: Fri, 16 Jan 2015 16:39:35 -0800
Subject: [PATCH 4/4] video: fbdev: Fix sparse warning messages

Use NULL instead of 0 for the last entry of dmt_modes struct.
Supresses "sparse: Using plain integer as NULL pointer" warning.

Signed-off-by: David Ung <davidu@nvidia.com>
Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ti.com>
---
 drivers/video/fbdev/core/modedb.c | 74 +++++++++++++++++++--------------------
 1 file changed, 37 insertions(+), 37 deletions(-)

diff --git a/drivers/video/fbdev/core/modedb.c b/drivers/video/fbdev/core/modedb.c
index 858a97e..7d07cf8 100644
--- a/drivers/video/fbdev/core/modedb.c
+++ b/drivers/video/fbdev/core/modedb.c
@@ -511,69 +511,69 @@ const struct dmt_videomode dmt_modes[DMT_SIZE] = {
 	{ 0x0a, 0x454c, 0x000000, &vesa_modes[9] },
 	{ 0x0b, 0x454f, 0x000000, &vesa_modes[10] },
 	{ 0x0c, 0x4559, 0x000000, &vesa_modes[11] },
-	{ 0x0d, 0x0000, 0x000000, 0 },
-	{ 0x0e, 0x0000, 0x000000, 0 },
+	{ 0x0d, 0x0000, 0x000000, NULL },
+	{ 0x0e, 0x0000, 0x000000, NULL },
 	{ 0x0f, 0x0000, 0x000000, &vesa_modes[12] },
 	{ 0x10, 0x6140, 0x000000, &vesa_modes[13] },
 	{ 0x11, 0x614a, 0x000000, &vesa_modes[14] },
 	{ 0x12, 0x614f, 0x000000, &vesa_modes[15] },
 	{ 0x13, 0x6159, 0x000000, &vesa_modes[16] },
-	{ 0x14, 0x0000, 0x000000, 0 },
+	{ 0x14, 0x0000, 0x000000, NULL },
 	{ 0x15, 0x714f, 0x000000, &vesa_modes[17] },
-	{ 0x16, 0x0000, 0x7f1c21, 0 },
-	{ 0x17, 0x0000, 0x7f1c28, 0 },
-	{ 0x18, 0x0000, 0x7f1c44, 0 },
-	{ 0x19, 0x0000, 0x7f1c62, 0 },
-	{ 0x1a, 0x0000, 0x000000, 0 },
-	{ 0x1b, 0x0000, 0x8f1821, 0 },
-	{ 0x1c, 0x8100, 0x8f1828, 0 },
-	{ 0x1d, 0x810f, 0x8f1844, 0 },
-	{ 0x1e, 0x8119, 0x8f1862, 0 },
-	{ 0x1f, 0x0000, 0x000000, 0 },
+	{ 0x16, 0x0000, 0x7f1c21, NULL },
+	{ 0x17, 0x0000, 0x7f1c28, NULL },
+	{ 0x18, 0x0000, 0x7f1c44, NULL },
+	{ 0x19, 0x0000, 0x7f1c62, NULL },
+	{ 0x1a, 0x0000, 0x000000, NULL },
+	{ 0x1b, 0x0000, 0x8f1821, NULL },
+	{ 0x1c, 0x8100, 0x8f1828, NULL },
+	{ 0x1d, 0x810f, 0x8f1844, NULL },
+	{ 0x1e, 0x8119, 0x8f1862, NULL },
+	{ 0x1f, 0x0000, 0x000000, NULL },
 	{ 0x20, 0x8140, 0x000000, &vesa_modes[18] },
 	{ 0x21, 0x8159, 0x000000, &vesa_modes[19] },
-	{ 0x22, 0x0000, 0x000000, 0 },
+	{ 0x22, 0x0000, 0x000000, NULL },
 	{ 0x23, 0x8180, 0x000000, &vesa_modes[20] },
 	{ 0x24, 0x818f, 0x000000, &vesa_modes[21] },
 	{ 0x25, 0x8199, 0x000000, &vesa_modes[22] },
-	{ 0x26, 0x0000, 0x000000, 0 },
-	{ 0x27, 0x0000, 0x000000, 0 },
-	{ 0x28, 0x0000, 0x000000, 0 },
-	{ 0x29, 0x0000, 0x0c2021, 0 },
-	{ 0x2a, 0x9040, 0x0c2028, 0 },
-	{ 0x2b, 0x904f, 0x0c2044, 0 },
-	{ 0x2c, 0x9059, 0x0c2062, 0 },
-	{ 0x2d, 0x0000, 0x000000, 0 },
-	{ 0x2e, 0x9500, 0xc11821, 0 },
-	{ 0x2f, 0x9500, 0xc11828, 0 },
-	{ 0x30, 0x950f, 0xc11844, 0 },
-	{ 0x31, 0x9519, 0xc11868, 0 },
-	{ 0x32, 0x0000, 0x000000, 0 },
+	{ 0x26, 0x0000, 0x000000, NULL },
+	{ 0x27, 0x0000, 0x000000, NULL },
+	{ 0x28, 0x0000, 0x000000, NULL },
+	{ 0x29, 0x0000, 0x0c2021, NULL },
+	{ 0x2a, 0x9040, 0x0c2028, NULL },
+	{ 0x2b, 0x904f, 0x0c2044, NULL },
+	{ 0x2c, 0x9059, 0x0c2062, NULL },
+	{ 0x2d, 0x0000, 0x000000, NULL },
+	{ 0x2e, 0x9500, 0xc11821, NULL },
+	{ 0x2f, 0x9500, 0xc11828, NULL },
+	{ 0x30, 0x950f, 0xc11844, NULL },
+	{ 0x31, 0x9519, 0xc11868, NULL },
+	{ 0x32, 0x0000, 0x000000, NULL },
 	{ 0x33, 0xa940, 0x000000, &vesa_modes[23] },
 	{ 0x34, 0xa945, 0x000000, &vesa_modes[24] },
 	{ 0x35, 0xa94a, 0x000000, &vesa_modes[25] },
 	{ 0x36, 0xa94f, 0x000000, &vesa_modes[26] },
 	{ 0x37, 0xa959, 0x000000, &vesa_modes[27] },
-	{ 0x38, 0x0000, 0x000000, 0 },
-	{ 0x39, 0x0000, 0x0c2821, 0 },
-	{ 0x3a, 0xb300, 0x0c2828, 0 },
-	{ 0x3b, 0xb30f, 0x0c2844, 0 },
-	{ 0x3c, 0xb319, 0x0c2868, 0 },
-	{ 0x3d, 0x0000, 0x000000, 0 },
+	{ 0x38, 0x0000, 0x000000, NULL },
+	{ 0x39, 0x0000, 0x0c2821, NULL },
+	{ 0x3a, 0xb300, 0x0c2828, NULL },
+	{ 0x3b, 0xb30f, 0x0c2844, NULL },
+	{ 0x3c, 0xb319, 0x0c2868, NULL },
+	{ 0x3d, 0x0000, 0x000000, NULL },
 	{ 0x3e, 0xc140, 0x000000, &vesa_modes[28] },
 	{ 0x3f, 0xc14f, 0x000000, &vesa_modes[29] },
-	{ 0x40, 0x0000, 0x000000, 0 },
+	{ 0x40, 0x0000, 0x000000, NULL},
 	{ 0x41, 0xc940, 0x000000, &vesa_modes[30] },
 	{ 0x42, 0xc94f, 0x000000, &vesa_modes[31] },
-	{ 0x43, 0x0000, 0x000000, 0 },
+	{ 0x43, 0x0000, 0x000000, NULL },
 	{ 0x44, 0x0000, 0x572821, &vesa_modes[34] },
 	{ 0x45, 0xd100, 0x572828, &vesa_modes[35] },
 	{ 0x46, 0xd10f, 0x572844, &vesa_modes[36] },
 	{ 0x47, 0xd119, 0x572862, &vesa_modes[37] },
-	{ 0x48, 0x0000, 0x000000, 0 },
+	{ 0x48, 0x0000, 0x000000, NULL },
 	{ 0x49, 0xd140, 0x000000, &vesa_modes[32] },
 	{ 0x4a, 0xd14f, 0x000000, &vesa_modes[33] },
-	{ 0x4b, 0x0000, 0x000000, 0 },
+	{ 0x4b, 0x0000, 0x000000, NULL },
 	{ 0x4c, 0x0000, 0x1f3821, &vesa_modes[38] },
 	{ 0x4d, 0x0000, 0x1f3828, &vesa_modes[39] },
 	{ 0x4e, 0x0000, 0x1f3844, &vesa_modes[40] },
-- 
2.3.1

